cmd/antrea-agent/options.go

```diff
@@ -78,6 +80,7 @@ func newOptions() *Options {
 // addFlags adds flags to fs and binds them to options.
 func (o *Options) addFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&o.configFile, "config", o.configFile, "The path to the configuration file")
+	fs.BoolVar(&o.windowsService, "service", false, "Run the daemon as a windows service")
```

Comment:
since you are adding some OS-specific code for initialization, could you also make sure that this flag is only added for Windows?

Reply:
FYI. - you can use runtime.IsWindowsPlatform() in antrea.io/antrea/pkg/util/runtime to check.